### PR TITLE
Extend fuzzinator.call.FileReaderDecorator with cleanup functionality

### PR DESCRIPTION
**Pull request by renatahodovan**

### CODE REVIEW

File: fuzzinator/call/file_reader_decorator.py [hunk 1]
```diff
@@ -43,4 +52,7 @@ def call(self, cls, obj, *, test, **kwargs):
                 issue['test'] = f.read()
             issue['filename'] = os.path.basename(test)
 
+        if self.cleanup:
+            os.remove(test)
```
Comment (akosthekiss) [inline fuzzinator/call/file_reader_decorator.py:56]: Can we have a test for this somehow? An input file copied into a temp directory and then checked for its existence after a call?

File: tests/call/test_file_reader_decorator.py [hunk 2]
```diff
@@ -7,22 +7,30 @@
...
 import fuzzinator
 
 from common_call import blinesep, resources_dir, MockAlwaysFailCall, MockNeverFailCall
 
 
-@pytest.mark.parametrize('call_init_kwargs, call_kwargs', [
-    ({'init_foo': 'init_bar'}, {'test': os.path.join(resources_dir, 'mock_tests', 'bar.txt')})
+@pytest.mark.parametrize('call_init_kwargs, call_kwargs, decorator_init_kwargs', [
+    ({'init_foo': 'init_bar'}, {'test': os.path.join(resources_dir, 'mock_tests', 'bar.txt')}, {'cleanup': False}),
+    ({'init_foo': 'init_bar'}, {'test': os.path.join(resources_dir, 'mock_tests', 'bar.txt')}, {'cleanup': True})
```
Comment (akosthekiss) [inline tests/call/test_file_reader_decorator.py:20]: Don't have to duplicate the entries here only to have cleanup with True and False. A new decorator can be stacked on top of the existing one and then pytest will create all the combinations. I.e.:

```py
@pytest.mark.parametrize('dec_kwargs', [
    {},
    {'cleanup': False},
    {'cleanup': True},
])
@pytest.mark.parametrize('call_init_kwargs, call_kwargs', [
    ({'init_foo': 'init_bar'}, {'test': os.path.join(resources_dir, 'mock_tests', 'bar.txt')})
])
```

Notes:
- the default of `cleanup` should also be tested, thus the empty dict
- `dec_kwargs` is the name used thoughout the tests

File: tests/call/test_file_reader_decorator.py [hunk 3]
```diff
@@ -7,22 +7,30 @@
...
 ])
 @pytest.mark.parametrize('call_class, exp', [
     (MockAlwaysFailCall, {'init_foo': 'init_bar', 'test': b'bar' + blinesep, 'filename': 'bar.txt'}),
     (MockNeverFailCall, None),
 ])
-def test_file_reader_decorator(call_class, call_init_kwargs, call_kwargs, exp):
-    call_class = fuzzinator.call.FileReaderDecorator()(call_class)
+def test_file_reader_decorator(call_class, call_init_kwargs, decorator_init_kwargs, call_kwargs, exp):
```
Comment (akosthekiss) [inline tests/call/test_file_reader_decorator.py:26]: The order of args used throughout the tests is `(call_class, call_init_kwargs, call_kwargs, dec_kwargs, exp, tmpdir)`. Adding `tmpdir` to the signature will also automatically create a temporary directory when running the test, this is handled by pytest, making `tempfile` superfluous.

File: tests/call/test_file_reader_decorator.py [hunk 4]
```diff
@@ -7,22 +7,30 @@
...
+    call_class = fuzzinator.call.FileReaderDecorator(**decorator_init_kwargs)(call_class)
     call = call_class(**call_init_kwargs)
 
-    with call:
-        assert call(**call_kwargs) == exp
+    with tempfile.TemporaryDirectory() as tempdir:
+        shutil.copy(call_kwargs['test'], tempdir)
+        test_path = os.path.join(tempdir, os.path.basename(call_kwargs['test']))
```
Comment (akosthekiss) [inline tests/call/test_file_reader_decorator.py:32]: `shutil.copy` returns the path to the newly created file, so this can be simplified to

```py
test_path = shutil.copy(call_kwargs['test'], tempdir)
```

File: tests/call/test_file_reader_decorator.py [hunk 5]
```diff
@@ -7,22 +7,30 @@
...
+
+        with call:
+            assert call(**dict(call_kwargs, test=test_path)) == exp
+            assert decorator_init_kwargs['cleanup'] == (not os.path.exists(test_path))
```
Comment (akosthekiss) [inline tests/call/test_file_reader_decorator.py:36]: To handle the default of `cleanup` properly, this should be `dec_kwargs.get('cleanup', False) == ...`

**Review by akosthekiss — CHANGES_REQUESTED** [review]

**Review by akosthekiss — APPROVED** [review]
lgtm
